include/libcyphal/transport/udp/transport.hpp

```diff
@@ -0,0 +1,99 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_UDP_TRANSPORT_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_UDP_TRANSPORT_HPP_INCLUDED
+
+#include "media.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/transport/multiplexer.hpp"
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace udp
+{
+
+class IUdpTransport : public ITransport
+{};
+
+namespace detail
+{
+
+class TransportImpl final : public IUdpTransport
+{
+public:
+    // MARK: IUpdTransport
+
+    // MARK: ITransport
+
+    CETL_NODISCARD cetl::optional<NodeId> getLocalNodeId() const noexcept override
+    {
+        return cetl::nullopt;
+    }
+    CETL_NODISCARD ProtocolParams getProtocolParams() const noexcept override
+    {
+        return ProtocolParams{};
+    }
+
+    CETL_NODISCARD Expected<UniquePtr<IMessageRxSession>, AnyError> makeMessageRxSession(
+        const MessageRxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IMessageTxSession>, AnyError> makeMessageTxSession(
+        const MessageTxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IRequestRxSession>, AnyError> makeRequestRxSession(
+        const RequestRxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IRequestTxSession>, AnyError> makeRequestTxSession(
+        const RequestTxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IResponseRxSession>, AnyError> makeResponseRxSession(
+        const ResponseRxParams&) override
+    {
+        return NotImplementedError{};
+    }
+    CETL_NODISCARD Expected<UniquePtr<IResponseTxSession>, AnyError> makeResponseTxSession(
+        const ResponseTxParams&) override
+    {
+        return NotImplementedError{};
+    }
+
+    // MARK: IRunnable
+
+    void run(const TimePoint) override {}
```

Comment:
prefer final to override. If actually using override document how subclasses must handle the base-class implementation (e.g. they shall-or-shall-not invoke it. They shall-invoke it first/last. etc).

Reply:
At class level we have `final` - afaik it means that all the virtual methods are final as well. do you want make it even more explicit?

Reply:
About "shall-or-shall-not invoke" I can't comment yet (due to lack of fine understanding of requirements), but I agree we should have (eventually) such documentation.

Reply:
Eh. I guess if the class is final the sure. It's fine.